### PR TITLE
Adding uid, load_fs, mknod, multiple_open, persistence setup, persistence test, rmdir, stat_complex, stat_mode tests and fixing zerofill fn

### PR DESCRIPTION
**Pull request by TristanB22**
Please take a closer look at the mknod call -- specifically the second portion. The repy test says 
```
assert lind_test_server.get_fscall_obj(-1).write_syscall(fd, "test") == 4,\
  "Write failed to /dev/random file."
assert lind_test_server.get_fscall_obj(-1).read_syscall(fd, 10) != '',\
  "Read failed from /dev/random file."
```
which doesn't really make sense to me. But, when I write 
```        
assert_eq!(cage.lseek_syscall(fd2, 0, SEEK_SET), 0);
assert_eq!(cage.write_syscall(fd2, str2cbuf("testing"), 7), 7);
assert_eq!(cage.lseek_syscall(fd2, 0, SEEK_SET), 0);
assert_ne!(cage.read_syscall(fd2, buf2.as_mut_ptr(), 7), 0);
```
it seems to work. I guess it is correct, but it confuses me. 

Also, the code in rmdir can be uncommented and run once Kaitlyn's PR is approved. It should work with her syscall provided the success code is 0.

With this PR, I can't write any more of the tests without more syscalls being implemented. Check [this](https://github.com/Lind-Project/safeposix-rust/issues/12) issue for a full synopsis.

### CODE REVIEW

File: src/tests/fs_tests.rs [hunk 1]
```diff
@@ -303,38 +378,401 @@ mod fs_tests {
...
+    }
+
+
+
+    pub fn ut_lind_fs_mknod() {
+        // let's create /dev/null
+        lindrustinit();
+        let cage = {CAGE_TABLE.read().unwrap().get(&1).unwrap().clone()};
+        let dev = makedev(&DevNo {major: 1, minor: 3});
+        let path = String::from("/null");
```
Comment (jesings) [inline src/tests/fs_tests.rs:628]: Again, same comment

File: src/tests/fs_tests.rs [hunk 2]
```diff
@@ -303,38 +378,401 @@ mod fs_tests {
...
+        let cage = {CAGE_TABLE.read().unwrap().get(&1).unwrap().clone()};
+        let path = String::from("/fooFileMode");
+        let fd = cage.open_syscall(&path, O_CREAT | O_EXCL | O_WRONLY, S_IRWXA);
+
+        let mut statdata = StatData::default();
+        assert_eq!(cage.stat_syscall(&path, &mut statdata), 0);
+        assert_eq!(statdata.st_mode, S_IRWXA | S_IFREG as u32);
+
+        //make a file without permissions and check that it is a reg file without permissions
+        let path2 = String::from("/fooFileMode2");
```
Comment (jesings) [inline src/tests/fs_tests.rs:768]: String::from one last time should be removed

**Review by jesings — COMMENTED** [review]
Aside from a few spot comments, my issues largely boil down to not asserting the result of stat enough and overusing String::from. First of all, it's more conventional to call "stringconst".to_string() to get a String from a &str literal, but second of all, the syscalls you're calling take a &str rather than a String, and you borrow that String  in each call to make an &String which coerces to a &str. It'd be better if you just assigned the &str literal to the variables and then didn't borrow the path in the call. 

For example:

```rs
let path = "/testpath";
cage.open_syscall(path, O_RDWR, 0);
```

**Comment by rennergade** [comment]
This looks good. Seems like you need to fix some merge conflicts though. Is this finalized?

**Pull request closed without merging**